### ISSUE TITLE
order: '..' is not recognized

### ISSUE THREAD

**Issue by novemberborn** [benmosher/eslint-plugin-import#1405]
https://github.com/benmosher/eslint-plugin-import/pull/1375 added support for "unknown" imports. Imports from `..` are classified as *unknown*. I think it ought to be *parent*.

```js
import foo from '../foo'
import { bar } from '..'
import baz from '.'
```

**Comment by mattrabe** [comment]
I agree. And `.` should be `index`. Otherwise, with the following eslint config:

```
    'import/order': [
      'error',
      {
        groups: [
          'builtin',
          'external',
          'internal',
          'parent',
          'sibling',
          'index',
        ],
        'newlines-between': 'always-and-inside-groups',
      },
    ],
```

`import/order` now wants to change:

```
import Form from '..'

import Checkbox from '.'
```

to:

```
import Checkbox from '.'

import Form from '..'
```

`../` and `./` are correctly recognized as `parent` and `index`, respectively, but `import/no-useless-path-segments` makes us change `../`to `..` and `./` to `.`, which now fail to be correctly recognized as OP states

**Comment by ljharb** [comment]
`.` should be index, unless there's a package.json in the folder, at which point it gets more complicated.

Mentioning that makes me realize that `../` also is only "parent" if it doesn't hit a package.json, or if it hits a package.json that resolves to a parent import. If package.json has "main" set to "lib/foo", and "lib/bar" imports "../", that's not a parent, it's a sibling.
